compiler/resolution/lowerIterators.cpp

```diff
@@ -57,9 +57,11 @@ FnSymbol* getTheIteratorFn(CallExpr* call) {
 FnSymbol* getTheIteratorFn(Type* icType)
 {
   // the asserts document the current state
+  AggregateType* icTypeAgg = toAggregateType(icType);
```

Comment:
I think I will handle that in a different PR, so I can merge this one today and focus on other things for a little while